src/app/src/components/FacilityListItemsTableSearch.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import { func, string } from 'prop-types';
+
+import CircularProgress from '@material-ui/core/CircularProgress';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import TextField from '@material-ui/core/TextField';
+import Search from '@material-ui/icons/Search';
+import Close from '@material-ui/icons/Close';
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_ESCAPE = 27;
+
+export default function FacilityListItemsTableSearch({
+    currentSearchTerm,
+    onSearch,
+}) {
+    const [isSearching, setSearching] = useState(false);
+    const [isCurrent, setCurrent] = useState(true);
+    const [searchText, setSearchText] = useState(currentSearchTerm);
+
+    const doSearch = (term) => {
+        setSearching(true);
+        onSearch(term);
+        setSearching(false);
+        setCurrent(true);
+    };
+
+    const handleSearch = () => {
+        if (isCurrent) {
+            return;
+        }
+
+        doSearch(searchText);
+    };
+
+    const handleTextChange = (e) => {
+        setSearchText(e.target.value);
+        setCurrent(false);
+    };
+
+    const handleReset = () => {
+        setSearchText('');
+        setCurrent(false);
+        doSearch('');
+    };
+
+    const handleTextKeyDown = (e) => {
+        if (e.keyCode === KEYCODE_ENTER) {
+            handleSearch();
+        } else if (e.keyCode === KEYCODE_ESCAPE) {
```

Comment:
We can probably keep these constants here but I noticed that the `constants` file already has a constant for the enter key to compare against `e.key`.

We also have this `makeSubmitFormOnEnterKeyPress` function -- https://github.com/open-apparel-registry/open-apparel-registry/blob/490d2866ffc7a72666299535dbc8611df2643715/src/app/src/util/util.js#L468 -- although it was designed to be used like this:

https://github.com/open-apparel-registry/open-apparel-registry/blob/312ccdbdb2659bafd952a180ee79d39abddac2e4/src/app/src/components/SendResetPasswordEmailForm.jsx#L146

Reply:
Ah thanks for the tip! I'll try to reuse that.